easybuild/main.py

```diff
@@ -219,9 +221,68 @@ def main(args=None, logfile=None, do_build=None, testing=False):
     eb_cmd_line = eb_go.generate_cmd_line() + eb_go.args
     log_start(eb_cmd_line, eb_tmpdir)
 
+    # report on umask here so we don't have to forward this info to `main`
     if options.umask is not None:
         _log.info("umask set to '%s' (used to be '%s')" % (oct(new_umask), oct(old_umask)))
 
+    # sanity check
+    if options.daemonize and options.logtostdout:
+        raise EasyBuildError(
+            "Option `--daemonize` is incompatible with `--logtostdout`."
+            " Please use only one of the two.")
+
+    if options.daemonize:
+        main_daemonize(
+            # local state
+            init_session_state, eb_go, eb_tmpdir, eb_cmd_line,
+            # original `main` args
+            args, logfile, do_build, testing)
+    else:
+        main_body(
+            # local state
+            init_session_state, eb_go, eb_tmpdir, eb_cmd_line,
+            # original `main` args
+            args, logfile, do_build, testing)
+
+
+def main_daemonize(init_session_state, eb_go, eb_tmpdir, eb_cmd_line,
+                   args=None, logfile=None, do_build=None, testing=False):
+    pidfile = os.path.join(eb_tmpdir, 'easybuild.pid')
+    print_msg("Backgrounding EasyBuild process,"
+              " will record PID in file '%s' ..." % (pidfile,))
+    from daemonize import Daemonize
+    daemon = Daemonize(
+        app='EasyBuild',
+        pid=pidfile,
+        chdir=os.getcwd(),
+        auto_close_fds=False,
+        logger=NonTerminatingLoggerProxy(_log),
+        verbose=True,
+        action=(lambda: main_body(
```

Comment:
> > +        action=(lambda: main_body(
>
> use `*args, **kwargs`?

No, the function passed as `action` is called by `daemon.start()` and is
called without any arguments.